### PR TITLE
matrix remove

### PR DESCRIPTION
**Pull request by diegoribero**
Adds `matrix remove` and `matrix remove <id>`, those commands allow the removal of devices.

The command will use the currently selected device or the one specified and will show a confirmation message before removing it if it does exist.

_Requires **matrix-firebase** `dr/remove` update._

### CODE REVIEW

**Comment by eighteyes** [comment]
Default to No please.

**Comment by eighteyes** [comment]
Can we support device name lookups too? Just like `use` ?

**Comment by eighteyes** [comment]
Merged the firebase branch, good work. :)

**Comment by eighteyes** [comment]
One more thing please add an entry to HISTORY.md

**Comment by diegoribero** [comment]
Updated according to comments @eighteyes :)

- Default prompt to NO
- Remove device from local cache on deletion
- Allow remove to receive a name or an Id
- Add changes to History
- Fix the error message when the uploadURL fails (Different topic but really simple fix)

File: bin/matrix-remove.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  if (showTheHelp) {
+    return displayHelp();
+  }
+
+  var targetValue, progress, target;
+  if (Matrix.pkgs.length < 1) { // No id specified
+    Matrix.validate.device(); // Require a selected device
+    target = Matrix.config.device.identifier;
+  } else if (Matrix.pkgs.length === 1) { //Id specified
+    target = Matrix.pkgs[0];
```
Comment (eighteyes) [inline bin/matrix-remove.js]: what if my name is > 1 word?

Comment (eighteyes) [inline bin/matrix-remove.js]: i'll just fix :)

**Pull request merged**